### ISSUE TITLE
Split XQuery extension modules into seperate builds

### ISSUE THREAD

**Issue by adamretter** [eXist-db/exist#805]
The XQuery extension modules under `$EXIST_HOME/extensions/modules` are at present built as a single Ant task. This has some disadvantages:

1. It is not clear where tests for these modules should go as there is just a `src/${package}` root which does not follow the standard java project layout. Also as modules in this single source root can be excluded from compilation via `$EXIST_HOME/extensions/modules/local.build.properties` this makes inclusion of module specific tests difficult.

2. It is not possible to describe the dependencies of each module independently. At the moment there is a single `ivy.xml` which is disabled which lists some dependencies of some modules but not all. We should have an `ivy.xml` per module.

We should instead make each module a separate build task with its own src root which follows a standard java project layout and has its own `ivy.xml` for its dependencies.

Prerequisite for writing tests for: https://github.com/eXist-db/exist/issues/790

**Comment by dizzzz** [comment]
you are right, but well we learned to live with it :-)  maybe not for 3.0 ?

**Comment by line-o** [comment]
This issue is no longer needed as the build tooling was switched to a multi-module maven project.
